omnigibson/systems/micro_particle_system.py

```diff
@@ -890,9 +765,9 @@ def generate_particle_instancer(
 
         # Automatically generate an identification number for this instancer if none is specified
         if idn is None:
-            idn = cls.max_instancer_idn + 1
-            # Also increment this counter
-            cls.max_instancer_idn += 1
+            idn = cls.next_available_instancer_idn
+
+        assert idn not in cls.instancer_idns, f"instancer idn {idn} already exists."
```

Comment:
this assert should maybe me moved to cls.next_available_instancer_idn?

Reply:
hmm I don't think so. `next_available_instancer_idn` is guaranteed to not exist. This is to safeguard if someone pass in an idn that already exists when calling `generate_particle_instancer`

Reply:
oh oh i see! makes sense